### ISSUE TITLE
Implementation of backside excitation condition

### ISSUE THREAD

**Issue by AvonReppert** [dschick/udkm1Dsim#131]
In some cases, experiments are carried out in a backside-pump, frontside probe geometry. *
It would be a quality of life upgrade to the workflow if udkm1Dsim could natively support this excitation scenario.

I'm thinking of a modification of the corresponding method in the `Heat` class

```
     get_absorption_profile(self, distances=[], backside_excitation = False ) 
      if ~backside_excitation:
          if self._excitation['multilayer_absorption']:
               dAdz, _, _, _ = self.get_multilayers_absorption_profile(distances)
              return dAdz
          else:
              return self.get_Lambert_Beer_absorption_profile(distances)
      else:
         if self._excitation['multilayer_absorption']:
               dAdz, _, _, _ = self.get_multilayers_absorption_profile_backside(distances)
              return dAdz
        else:
              return self.get_Lambert_Beer_absorption_profile_backside(distances)

```

where the corresponding functions `get_Lambert_Beer_absorption_profile_backside` and `get_multilayers_absorption_profile_backside` have to be introduced to the `Heat` class based on the reversed structure. I am not sure if the backside excitation would affect the workflow somewhere else in the code but I guess not. I hope this should be the least intrusive way to introduced this feature while being backwards compatible with existing script. 

*Magneto-accoustic experiments or experiments in nonlinear phononics for example, want to separate the excitation region from the probed region and therefore often turn to backside excitation geometry while probing at the frontside of an opaque medium. 

When implemented it should be possible to attain the following penetration profiles out of the box
![backside_excitation and frontside_excitation](https://github.com/dschick/udkm1Dsim/assets/36692180/3f774d95-cbfe-448d-8b3d-2a391255c274)
Using these parameters:
![image](https://github.com/dschick/udkm1Dsim/assets/36692180/9549ac76-5003-4d17-98c7-f37d4f916308)
I just add this here as a reference for comparison

**Comment by dschick** [comment]
Hi @AvonReppert ,

that is a pretty good idea, and I think it should not be too hard to implement.
I would suggest adding an additional boolean parameter `backside_excitation` to the `heat.excitation` dictionary:
```python
h.excitation = {'fluence': [5]*u.mJ/u.cm**2,
                'delay_pump':  [0]*u.ps,
                'pulse_width':  [0]*u.ps,
                'multilayer_absorption': True,
                'wavelength': 800*u.nm,
                'theta': 45*u.deg,
                'backside_excitation' True,
                }
```
which will then be handed to the according methods for calculating the absorption profile.

**Comment by dschick** [comment]
Hi @AvonReppert 

you check the backside excitation in the PR #132 and the according branch.

Let me know what you think

**Comment by dschick** [comment]
For now, Lambert Beer does not take the substrate into account for backside pumping and we should discuss what is the propper way to do so also for the MultilayerAbsorption.

**Comment by dschick** [comment]
The current PR #132 is ready for testing.
Please let me know if everything works as expected.

**Comment by AvonReppert** [comment]
Thanks this is  great!
I tested it, it works  as expected and yields the same results as for manually building an inverted sample structure. 

```
dAdz_frontside = h.get_absorption_profile([], backside=False)
dAdz_backside = h.get_absorption_profile([], backside=True)
```


yields the right profiles:
![profile](https://github.com/dschick/udkm1Dsim/assets/36692180/2453df33-af1c-4e0c-b5b6-f7672c3ff74c)

also the corresponding heat diffusion works as expected:
![results](https://github.com/dschick/udkm1Dsim/assets/36692180/47a7582c-37f9-4b5d-b470-72348e5ebd1b)

using this command:

```
h = ud.Heat(S, True)
h.excitation = {'fluence':    [fluencepump]*u.mJ/u.cm**2,
                'delay_pump':  [0]*u.ps,
                'pulse_width': [0.]*u.ps,
                'multilayer_absorption': True,
                'wavelength': 800*u.nm,
                'theta': 90*u.deg,
                'backside': True}
```

**Comment by dschick** [comment]
issue solved via merging PR #132
